tests/MongoDB.Bson.TestHelpers/JsonDrivenTests/JsonDrivenTestCaseFactory.cs

```diff
@@ -19,43 +19,30 @@
 using System.IO;
 using System.Linq;
 using System.Reflection;
+using MongoDB.Bson.IO;
+using MongoDB.Bson.Serialization;
+using MongoDB.Bson.Serialization.Serializers;
 using Xunit.Abstractions;
 
 namespace MongoDB.Bson.TestHelpers.JsonDrivenTests
 {
-    public abstract class JsonDrivenTestCaseFactory<TTestCase> : IEnumerable<object[]>
-        where TTestCase : IXunitSerializable
+    public abstract class JsonTestReader
```

Comment:
I was initially suspicious about this, but it's starting to make sense.

If you keep this it needs to be moved to a file of its own.

Perhaps the name needs fixing too.

Seems like this is for reading arbitrary JSON files, not just `Test files.

Reply:
Actually... the most accurate name is probably `EmbeddedResourceJsonFileReader` or something like that.

Seems like the refactoring you have done here is to isolate the part of the code that reads JSON files from embedded resources.